service/controller/resource/heartbeat/resource.go

```diff
@@ -0,0 +1,102 @@
+package heartbeat
+
+import (
+	"fmt"
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/client"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/heartbeat"
+	"github.com/opsgenie/opsgenie-go-sdk-v2/og"
+
+	"github.com/giantswarm/prometheus-meta-operator/service/key"
+)
+
+const (
+	Name = "heartbeat"
+)
+
+type Config struct {
+	Logger       micrologger.Logger
+	Installation string
+	OpsgenieKey  string
+}
+
+type Resource struct {
+	logger          micrologger.Logger
+	heartbeatClient *heartbeat.Client
+	installation    string
+	opsgenieKey     string
+}
+
+func New(config Config) (*Resource, error) {
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.Installation == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Installation must not be empty", config)
+	}
+	if config.OpsgenieKey == "" {
+		return nil, microerror.Maskf(invalidConfigError, "%T.OpsgenieKey must not be empty", config)
+	}
+
+	c := &client.Config{
+		ApiKey:         config.OpsgenieKey,
+		OpsGenieAPIURL: client.API_URL,
+		RetryCount:     1,
+	}
+	client, err := heartbeat.NewClient(c)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	r := &Resource{
+		logger:          config.Logger,
+		heartbeatClient: client,
+		installation:    config.Installation,
+	}
+
+	return r, nil
+}
+
+func (r *Resource) Name() string {
+	return Name
+}
+
+func toHeartbeat(v interface{}, installation string) (*heartbeat.Heartbeat, error) {
+	cluster, err := key.ToCluster(v)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	name := fmt.Sprintf("%s-%s", installation, cluster.GetName())
```

Comment:
niceee

Reply:
small things that matters :)

I though it would be convenient to know which installation this is coming from by just looking at the name.